### PR TITLE
chore: Adds go vet analyzers in CI 

### PR DESCRIPTION
**Pull request by patriciareinoso**
Enable all go vet analyzers in CI
Fixes  `nilness` and `fieldalignment`  check

### CODE REVIEW

**Comment by gab-arrobo** [comment]
ok to test

File: factory/config.go [hunk 1]
```diff
@@ -47,12 +47,12 @@ type Configuration struct {
 }
 
 type Sbi struct {
+	Tls          *Tls   `yaml:"tls,omitempty"`
 	Scheme       string `yaml:"scheme"`
 	RegisterIPv4 string `yaml:"registerIPv4,omitempty"` // IP that is registered at NRF.
 	// IPv6Addr string `yaml:"ipv6Addr,omitempty"`
```
Comment (gab-arrobo) [inline factory/config.go:53]: @thakurajayL, should we remove this line (and other commented out code that is related to IPv6? Or are we planning to enable IPv6 support in the future?

File: httpcallback/router.go [hunk 2]
```diff
@@ -70,16 +70,11 @@ func Index(c *gin.Context) {
...
+		Name: "Index", Method: "GET",
+		Pattern: "/", HandlerFunc: Index,
 	},
-
 	{
-		"DataChangeNotificationToNF",
-		strings.ToUpper("Post"),
-		"/sdm-subscriptions",
-		HTTPDataChangeNotificationToNF,
+		Name: "DataChangeNotificationToNF", Method: strings.ToUpper("Post"),
```
Comment (gab-arrobo) [inline httpcallback/router.go:77]: Why not just type `POST` as is done in the previous set when using `GET`?
```suggestion
		Name: "DataChangeNotificationToNF", Method: "POST",
```

**Review by gab-arrobo — COMMENTED** [review]
Have you run an end-to-end test to make sure that proposed changes do not break anything?

File: factory/config.go (hunk 1 above)
Comment (thakurajayL) [reply factory/config.go:53]: We can remove it. We can bring it back when we wish to enable.

Comment (gab-arrobo) [reply factory/config.go:53]: @patriciareinoso, let's remove stuff related to IPv6 that is commented out, ok? We can do it in a another PR

**Review by gab-arrobo — APPROVED** [review]
+1 (tested changes with AiaB)

**Pull request merged**